src/utils/auth_utils.js

```diff
@@ -22,22 +28,56 @@ const firebaseConfig = {
 const app = initializeApp(firebaseConfig);
 const auth = getAuth(app);
 
-const signInWithGoogle = async () => {
-  // TODO
-};
-
 const logInWithEmailAndPassword = async (email, password) => {
   try {
     await signInWithEmailAndPassword(auth, email, password);
     alert('Sign in success');
   } catch (err) {
-    console.error(err);
     alert(err.message);
   }
 };
 
-const registerWithEmailAndPassword = async () => {
-  // TODO
+const createUserInFirebase = async (email, password) => {
+  try {
+    const user = await createUserWithEmailAndPassword(auth, email, password);
+    sendEmailVerification(user.user);
+    return user;
+  } catch (err) {
+    alert(err.message);
+    return err;
+  }
+};
+
+const createUserInDB = async userObject => {
+  try {
+    // Replace line below with call to NPO DB
+    // const res = await WMKBackend.post('/register/create', userObject);
+    // return res;
+  } catch (err) {
+    const { email, password } = userObject;
+
+    // since this route is called after user is created in firebase, if this
+    // route errors out, that means we have to discard the created firebase object
+    await signInWithEmailAndPassword(auth, email, password);
+    const userToBeTerminated = await auth.currentUser;
+    userToBeTerminated.delete();
+
+    throw new Error(err);
+  }
+};
+
+const signInWithGoogle = () => {};
+
+const registerWithEmailAndPassword = async (email, password) => {
+  try {
+    createUserInFirebase(email, password);
+    createUserInDB({ email, password });
```

Comment:
We combined the two functions in a createUser function. However, the signInWithEmailAndPassword is there in case there is an error when trying to create a user in the DB. In the case this errors out, we want to sign in the user and then delete them from firebase. Let us know if this looks good